src/TestLisExpr.chpl

```diff
@@ -62,19 +62,109 @@ module TestLisExpr
         return f;
     }
 
+     // tasks per locale based on locale0
+    const numTasks = here.maxTaskPar;
+    const Tasks = {0..#numTasks}; // these need to be const for comms/performance reasons
+    
+    // calculate sub-domain for task
+    inline proc calcBlock(task: int, low: int, high: int) {
+        var totalsize = high - low + 1;
+        var div = totalsize / numTasks;
+        var rem = totalsize % numTasks;
+        var rlow: int;
+        var rhigh: int;
+        if (task < rem) {
+            rlow = task * (div+1) + low;
+            rhigh = rlow + div;
+        }
+        else {
+            rlow = task * div + rem + low;
+            rhigh = rlow + div - 1;
+        }
+        return {rlow .. rhigh};
+    }
+
+    /* eval with a specific environment for the test expression return true if no errors */
+    proc test_eval_coforall(const in prog: string): bool {
+        var f = true;
+        var N = 10;
+        var D = {0..#N};
+        var A: [D] int = D;
+        var B: [D] int;
+        
+        // this could have the advantage of not creating array temps like the rest of arkouda does
+        try {
+            coforall loc in Locales {
+                on loc {
+                    coforall task in Tasks {
+                        // get local domain's indices
+                        var lD = A.domain.localSubdomain();
+                        // calc task's indices from local domain's indices
+                        var tD = calcBlock(task, lD.low, lD.high);
+                        
+                        // parse and check the program
+                        var ast = parse(prog);
+                        // allocate the env for variables
+                        var env = new owned Env();
+                        
+                        for i in tD {
+                            // addEnrtry redefines values for already existing entries
+                            env.addEntry("elt",A[i]); // add a symbol called "elt" and value for a
+                            
+                            // this version does the eval the in the enviroment which creates the symbol "ans"
+                            //var ans = env.lookup("ans").toValue(int).v; // retrieve value for ans
+                            //b = ans;
+                            
+                            // this version just returns the GenValue from the eval call
+                            var ans = eval(ast,env);
+                            B[i] = ans.toValue(int).v; // put answer into b
+                        }
+                    }
+                }
+                
+                writeln(A);
+                writeln(B);
+            }
+        }
+        catch errors: TaskErrors {
+            for e in errors {
+                // why does compile complain about not having a try her and not in the serial case???!
```

Comment:
@bradcray @bmcdonald3 why does compiler complain about not having a `try!` here and not in the serial case???!

Reply:
Hmm, I was able to remove the `try!` here without any problems on both 1.26 and 1.27 (see the snippet below).